code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,36 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("- no Deepmaint Z-levels registered!")
```

Comment:
This log is lacking some context and might be out of place for other staff. Would a simple `to_chat` with error span class be more suitable?

Reply:
IIRC the method logs what proc was called by whom; it's a weird API

Reply:
Also this follows the convention of other Template-adjacent stuff IIRC

Reply:
Normal logs do not include calliing procs, liines etc. Only `WARNING`, `crash_with` and `ARGS_DEBUG` insert the calling proc and line automatically.
`log_and_message_admins` just inserts the calling user/mob with the `(?)` href link